### PR TITLE
JAMES-1794 Handle attachments in alternative part

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java [hunk 1]
```diff
@@ -60,7 +60,7 @@
                 .getBody();
         try {
             if (body instanceof Multipart) {
-                return listAttachments((Multipart)body);
+                return listAttachments((Multipart)body, Context.fromSubType(((Multipart)body).getSubType()));
```
Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java]: You may extract the cast of the body in a method.

Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java]: I don't think it's worth a method, but I can extract a variable if you want.

Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java]: done

**Comment by aduprat** [comment]
:+1:

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java [hunk 2]
```diff
@@ -169,9 +169,51 @@ public Boolean apply(String dispositionType) {
...
+        }
+
+        public static Context fromSubType(String subPart) {
+            if (isMainBody(subPart)) {
+                return BODY;
+            }
+            return OTHER;
+        }
+
+        private static boolean isMainBody(Entity entity) {
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java]: I don' really understand what you mean by "main body" ?

Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java]: The meaning has been lost in a refactoring, changed.

**Comment by aduprat** [comment]
:+1:

**Comment by mbaechler** [comment]
:+1:

**Pull request merged**